packages/flutter/lib/src/material/material_localizations.dart

```diff
@@ -696,24 +694,28 @@ class DefaultMaterialLocalizations implements MaterialLocalizations {
   }
 
   @override
-  DateTime parseCompactDate(String inputString) {
+  DateTime? parseCompactDate(String? inputString) {
+    if (inputString == null) {
+      return null;
+    }
```

Comment:
Why do we need to newly allow null here?

Reply:
Same cause : The call chain comes from `typedef FormFieldValidator<T> = String? Function(T? value);` where value is nullable.